packages/editor/src/components/widgets/table/row/ReorderRow.tsx

```diff
@@ -37,22 +48,40 @@ export const ReorderRow = ({ id, updateOrder, children, ...props }: ReorderRowPr
       } else {
         console.log(`invalid drop item ${e.items[0]}`);
       }
+      if (!row.getIsSelected()) {
+        row.getToggleSelectedHandler()(e);
+      }
     }
   });
 
   return (
-    <MessageRowWithTr
-      {...dragProps}
-      {...dropProps}
-      ref={ref}
-      colSpan={2}
-      className={`dnd-row${isDragging ? ' dragging' : ''}${isDropTarget ? ' target' : ''}`}
-      {...props}
-    >
-      {children}
-      <td className='dnd-row-handle'>
-        <IvyIcon icon={IvyIcons.ChangeType} />
-      </td>
-    </MessageRowWithTr>
+    <>
+      <SelectRow
+        {...dragProps}
+        {...dropProps}
+        row={row}
+        onDoubleClick={onDoubleClick}
+        onClick={e => {
+          if (!row.getIsSelected()) {
+            row.getToggleSelectedHandler()(e);
+          }
+        }}
+        title={title}
+        className={`dnd-row${isDragging ? ' dragging' : ''}${isDropTarget ? ' target' : ''} ${styleMessageRow(message)}`}
+        {...props}
+      >
+        {children}
+      </SelectRow>
+      <MessageRow colSpan={colSpan ? colSpan : 2} message={message} />
+    </>
   );
 };
+
+export const ReorderWrapperIcon = ({ children }: React.HTMLAttributes<HTMLTableCellElement>) => (
+  <div className='reorder-select-cell'>
```

Comment:
I think i would not call this 'cell' as it is not really a cell, I think I would call it 'reorder-select-icon' or 'reorder-select-handle'